lib/ReactViews/FeatureInfo/FeatureInfoSection.jsx

```diff
@@ -371,4 +377,79 @@ function describeFromProperties(properties, time) {
     return html;
 }
 
+/**
+ * @param  {ReactClass} that The FeatureInfoSection.
+ * @return {Object} Returns {templateData, info, rawData, showRawData, hasRawData}.
+ *                  templateData is the object passed to the templating engine.
+ *                  info is the main body of the info section, as a react component.
+ *                  rawData is the same for the raw data, if it needs to be shown.
+ *                  showRawData is whether to show the rawData.
+ *                  hasRawData is whether there is any rawData to show.
+ */
+function getInfoAsReactComponent(that) {
+    const templateData = that.getPropertyValues();
+    const updateCounters = that.props.feature.updateCounters;
+    if (defined(templateData)) {
+        delete templateData._terria_columnAliases;
+    }
+    const context = {
+        catalogItem: that.props.catalogItem,
+        feature: that.props.feature,
+        updateCounters: updateCounters
+    };
+    const showRawData = !that.hasTemplate() || that.state.showRawData;
+    let rawDataHtml;
+    let rawData;
+    if (showRawData) {
+        rawDataHtml = that.descriptionFromFeature();
+        if (defined(rawDataHtml)) {
+            rawData = renderMarkdownInReact(rawDataHtml, context);
+        }
+    }
+    return {
+        templateData: templateData,
+        info: that.hasTemplate() ? renderMarkdownInReact(that.descriptionFromTemplate(), context) : rawData,
+        rawData: rawData,
+        showRawData: showRawData,
+        hasRawData: !!rawDataHtml
+    };
+}
+
+function setTimeoutsForUpdatingCustomComponents(that) { // eslint-disable-line require-jsdoc
+    const {info} = getInfoAsReactComponent(that);
+    const foundCustomComponents = CustomComponents.find(info);
+    foundCustomComponents.forEach((match, componentNumber) => {
+        const updateSeconds = match.type.selfUpdateSeconds(match.reactComponent);
+        if (updateSeconds > 0) {
+            setTimeoutForUpdatingCustomComponent(that, match.reactComponent, updateSeconds, componentNumber);
+        }
+    });
+}
+
+function setTimeoutForUpdatingCustomComponent(that, reactComponent, updateSeconds, componentNumber) { // eslint-disable-line require-jsdoc
```

Comment:
Re logic in the UI - Note this piece only handles self-updating `<chart>` tags (ie. point 2 in my description at the top of this PR). These are only rendered when you've opened a Feature Info Section, and they don't actually change any of the feature's properties or description - so `definitionChanged` would never get triggered. So I saw <chart> updates as updates to the UI, not the feature. Is that plausible?

Using `definitionChanged` might simplify the implementation of point 1b in my description, though.